.github/workflows/add-label-when-promoted.yaml

```diff
@@ -0,0 +1,37 @@
+name: Check if commits are promoted
+
+on: [push]
+
+jobs:
+  check-commit:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+      issues: write
+    if: github.ref == 'refs/heads/master'
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Fetch all history for all tags and branches
+
+      - name: Label PRs
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          # Get the list of the latest commits
+          commits=$(git log --format=%B ${{ github.event.before }}..${{ github.event.after }})
+
+          echo "$commits" | while read line ; do
+            if [[ $line =~ Closes\ .*#([0-9]+) ]]; then
```

Comment:
Please move this to a python script and call it from here (assuming we can reference other files). Shell is not robust enough for this.

Reply:
@avikivity changed, added it under `scripts` folder